lib/block-supports/block-style-variations.php

```diff
@@ -486,6 +486,13 @@ function gutenberg_register_block_style_variations_from_theme_json_data( $variat
  * @access private
  */
 function gutenberg_register_block_style_variations_from_theme() {
+	$has_partials_directory = is_dir( get_stylesheet_directory() . '/styles' ) || is_dir( get_template_directory() . '/styles' );
+
+	// Skip any registration of styles if no theme.json or variation partials.
+	if ( ! wp_theme_has_theme_json() && ! $has_partials_directory ) {
+		return;
+	}
+
```

Comment:
Is this meant to be an `||` instead of an `&&` for the return?

Also, a tiny optimisation, but if it's meant to be an `||` we could do two separate checks, so that we return if `! wp_theme_has_theme_json()` and then return if `! $has_partials_directory`. That way there's no need to check if a directory exists if we know there's no theme JSON?

Reply:
Good question 👍 

The `||` was deliberate as a block theme may or may not have a theme.json file. Hybrid themes also complicate the situation. 

I've pushed a more detailed explanatory inline comment for this early return.

```
	/*
	 * Skip any registration of styles if no theme.json or variation partials are present.
	 *
	 * Given the possibility of hybrid themes, this check can't rely on if the theme
	 * is a block theme or not. Instead:
	 *   - If there is a primary theme.json, continue.
	 *   - If there is a partials directory, continue.
	 *   - The only variations to be registered from the global styles user origin,
	 *     are those that have been copied in from the selected theme style variation.
	 *     For a theme style variation to be selected it would have to have a partial
	 *     theme.json file covered by the previous check.
	 */
```

Reply:
Looks good, thanks for adding the explanatory comment!